tests/predict/test_predict.py

```diff
@@ -684,3 +686,42 @@ def test_predicted_outputs_piped_from_predict_to_lm_call():
         program(question="Why did a chicken cross the kitchen?", prediction="To get to the other side!")
 
     assert "prediction" not in mock_completion.call_args[1]
+
+
+def test_pydantic_special_type_serialization():
+    class WebsiteInfo(BaseModel):
+        name: str
+        url: HttpUrl
```

Comment:
actually I don't know `HttpUrl` is not json serializable until this issue report, maybe add a comment or change to well-known troublemakers like `datetime.datetime`?

Reply:
Actually do we support datetime as part of signature? I thought we only support primitives and pydantic